packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/ModelTypeProvider.cs

```diff
@@ -177,31 +104,26 @@ private IReadOnlyList<Parameter> BuildConstructorParameters(bool isSerialization
             foreach (var property in _inputModel.Properties)
             {
                 CSharpType propertyType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(property.Type);
-                var initializationValue = GetPropertyInitializationValue(property, propertyType);
-                var parameterValidation = GetParameterValidation(property, propertyType);
-
-                var parameter = new Parameter(
-                   Name: property.Name.ToVariableName(),
-                   Description: FormattableStringHelpers.FromString(property.Description),
-                   Type: propertyType,
-                   DefaultValue: null,
-                   Validation: parameterValidation,
-                   Initializer: null);
-
                 // All properties should be included in the serialization ctor
                 if (isSerializationConstructor)
                 {
-                    constructorParameters.Add(parameter with { Validation = ValidationType.None });
+                    constructorParameters.Add(new Parameter(property)
+                    {
+                        Validation = ValidationType.None,
+                    });
                 }
                 else
                 {
                     // For classes, only required + not readonly + not initialization value + not discriminator could get into the public ctor
                     // For structs, all properties must be set in the public ctor
-                    if (_isStruct || (property is { IsRequired: true, IsDiscriminator: false } && initializationValue == null))
+                    if (_isStruct || (property is { IsRequired: true, IsDiscriminator: false } && !propertyType.IsLiteral))
                     {
                         if (!property.IsReadOnly)
                         {
-                            constructorParameters.Add(parameter with { Type = parameter.Type.InputType });
+                            constructorParameters.Add(new Parameter(property)
+                            {
+                                Type = propertyType.InputType,
```

Comment:
This happens inside the ctor already.  I think you can remove `propertyType` from the scope here and it should clean up the logic a bit.

Reply:
ah it looks like for a model's public ctor, we use a different type for params that are lists or readonlymemory. Looks like we are using the `CSharpType`'s `InputType` here when constructing the params, and the serialization constructor will just use the default csharptype for it's ctor params (which is what is being done within the Parameter constructor).